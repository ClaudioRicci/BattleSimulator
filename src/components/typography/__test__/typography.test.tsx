import React from "react";
import ReactDOM from "react-dom";
import {
  H1Tag,
  H1TagGreen,
  H1TagRed,
  H2Tag,
  H3Tag,
  H4Tag
} from "../../typography";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

describe("Typography Tests", () => {
  describe("H1Tag Tests", () => {
    it("H1 Tag renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<H1Tag />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("H1 is primary colour (Navy Blue, #0000BD)", () => {
      const tree = renderer.create(<H1Tag />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("color", "#00008D");
    });
  });

  describe("H1TagGreen Tests", () => {
    it("H1TagGreen renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<H1TagGreen />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("H1TagGreen is green (#90EE90)", () => {
      const tree = renderer.create(<H1TagGreen />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("color", "#90EE90");
    });
  });

  describe("H1TagRed Tests", () => {
    it("H1TagRed renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<H1TagRed />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("H1TagRed is red (#FB2020)", () => {
      const tree = renderer.create(<H1TagRed />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("color", "#FB2020");
    });
  });

  describe("H2Tag Tests", () => {
    it("H2Tag renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<H2Tag />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("H1Tag is secondary colour (Neon Purple, #F900F1)", () => {
      const tree = renderer.create(<H2Tag />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("color", "#F900F1");
    });
  });

  describe("H3Tag Tests", () => {
    it("H3Tag renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<H3Tag />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("H3Tag is bright colour (Bright Red, #FB2020)", () => {
      const tree = renderer.create(<H3Tag />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("color", "#FB2020");
    });
  });

  describe("H4Tag Tests", () => {
    it("H4Tag renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<H4Tag />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("H4Tag is secondary colour (Neon Purple, #F900F1)", () => {
      const tree = renderer.create(<H4Tag />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("color", "#F900F1");
    });
  });
});
