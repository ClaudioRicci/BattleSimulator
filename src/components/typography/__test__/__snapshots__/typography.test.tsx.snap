// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Tests H1Tag Tests H1 is primary colour (Navy Blue, #0000BD) 1`] = `
.c0 {
  font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  color: #00008D;
  text-align: center;
  border-bottom: 0.2rem #00008D solid;
  padding-bottom: 0.2rem;
}

@media (max-width:768px) {
  .c0 {
    font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  }
}

<h1
  className="c0"
/>
`;

exports[`Typography Tests H1Tag Tests H1 is primary colour 1`] = `
.c0 {
  font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  color: #00008D;
  text-align: center;
  border-bottom: 0.2rem #00008D solid;
  padding-bottom: 0.2rem;
}

@media (max-width:768px) {
  .c0 {
    font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  }
}

<h1
  className="c0"
/>
`;

exports[`Typography Tests H1TagGreen Tests H1TagGreen is green (#0F0) 1`] = `
.c0 {
  font: 900 3em/1rem Verdana,Arial,Helvetica,sans-serif;
  color: #0f0;
}

@media (max-width:768px) {
  .c0 {
    font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  }
}

<h1
  className="c0"
/>
`;

exports[`Typography Tests H1TagGreen Tests H1TagGreen is green (#0f0) 1`] = `
.c0 {
  font: 900 3em/1rem Verdana,Arial,Helvetica,sans-serif;
  color: #0f0;
}

@media (max-width:768px) {
  .c0 {
    font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  }
}

<h1
  className="c0"
/>
`;

exports[`Typography Tests H1TagGreen Tests H1TagGreen is green 1`] = `
.c0 {
  font: 900 3em/1rem Verdana,Arial,Helvetica,sans-serif;
  color: #0f0;
}

@media (max-width:768px) {
  .c0 {
    font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  }
}

<h1
  className="c0"
/>
`;

exports[`Typography Tests H1TagGreen Tests H1TagGreen is secondary colour 1`] = `
.c0 {
  font: 900 3em/1rem Verdana,Arial,Helvetica,sans-serif;
  color: #0f0;
}

@media (max-width:768px) {
  .c0 {
    font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  }
}

<h1
  className="c0"
/>
`;

exports[`Typography Tests H1TagRed Tests H1TagRed is red (#F00) 1`] = `
.c0 {
  font: 900 3em/1rem Verdana,Arial,Helvetica,sans-serif;
  color: #f00;
}

@media (max-width:768px) {
  .c0 {
    font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  }
}

<h1
  className="c0"
/>
`;

exports[`Typography Tests H1TagRed Tests H1TagRed is red (#f00) 1`] = `
.c0 {
  font: 900 3em/1rem Verdana,Arial,Helvetica,sans-serif;
  color: #f00;
}

@media (max-width:768px) {
  .c0 {
    font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  }
}

<h1
  className="c0"
/>
`;

exports[`Typography Tests H1TagRed Tests H1TagRed is red 1`] = `
.c0 {
  font: 900 3em/1rem Verdana,Arial,Helvetica,sans-serif;
  color: #f00;
}

@media (max-width:768px) {
  .c0 {
    font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  }
}

<h1
  className="c0"
/>
`;

exports[`Typography Tests H1TagRed Tests H1TagRed is secondary colour 1`] = `
.c0 {
  font: 900 3em/1rem Verdana,Arial,Helvetica,sans-serif;
  color: #f00;
}

@media (max-width:768px) {
  .c0 {
    font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  }
}

<h1
  className="c0"
/>
`;

exports[`Typography Tests H2Tag Tests H1 is secondary colour 1`] = `
.c0 {
  font: 900 1.1em/1.1rem Verdana,Arial,Helvetica,sans-serif;
  color: #F900F1;
  line-height: 1rem;
}

<h2
  className="c0"
/>
`;

exports[`Typography Tests H2Tag Tests H1Tag is secondary colour (Neon Purple, #F900F1) 1`] = `
.c0 {
  font: 900 1.1em/1.1rem Verdana,Arial,Helvetica,sans-serif;
  color: #F900F1;
  line-height: 1rem;
}

<h2
  className="c0"
/>
`;

exports[`Typography Tests H2Tag Tests H1Tag is secondary colour 1`] = `
.c0 {
  font: 900 1.1em/1.1rem Verdana,Arial,Helvetica,sans-serif;
  color: #F900F1;
  line-height: 1rem;
}

<h2
  className="c0"
/>
`;

exports[`Typography Tests H3Tag Tests H1 is secondary colour 1`] = `
.c0 {
  font: 900 1.8em/1.8rem Verdana,Arial,Helvetica,sans-serif;
  color: #FB2020;
  line-height: 1rem;
}

<h3
  className="c0"
/>
`;

exports[`Typography Tests H3Tag Tests H3Tag is secondary colour 1`] = `
.c0 {
  font: 900 1.8em/1.8rem Verdana,Arial,Helvetica,sans-serif;
  color: #FB2020;
  line-height: 1rem;
}

<h3
  className="c0"
/>
`;

exports[`Typography Tests H3Tag Tests H3Tag is ternary colour (Bright Red, #FB2020) 1`] = `
.c0 {
  font: 900 1.8em/1.8rem Verdana,Arial,Helvetica,sans-serif;
  color: #FB2020;
  line-height: 1rem;
}

<h3
  className="c0"
/>
`;

exports[`Typography Tests H4Tag Tests H1 is secondary colour 1`] = `
.c0 {
  font: 900 0.8em/0.8rem Verdana,Arial,Helvetica,sans-serif;
  color: #F900F1;
  line-height: 1rem;
}

<h4
  className="c0"
/>
`;

exports[`Typography Tests H4Tag Tests H4Tag is secondary colour (Neon Purple, #F900F1) 1`] = `
.c0 {
  font: 900 0.8em/0.8rem Verdana,Arial,Helvetica,sans-serif;
  color: #F900F1;
  line-height: 1rem;
}

<h4
  className="c0"
/>
`;

exports[`Typography Tests H4Tag Tests H4Tag is secondary colour 1`] = `
.c0 {
  font: 900 0.8em/0.8rem Verdana,Arial,Helvetica,sans-serif;
  color: #F900F1;
  line-height: 1rem;
}

<h4
  className="c0"
/>
`;
